pkg/dockerregistry/server/auth.go

```diff
@@ -201,6 +196,11 @@ func TokenRealm(options map[string]interface{}) (*url.URL, error) {
 func newAccessController(options map[string]interface{}) (registryauth.AccessController, error) {
 	log.Info("Using Origin Auth handler")
 
+	ctx, ok := options["_context"].(context.Context)
```

Comment:
I don't think the context makes sense in functions where you don't deal with a request that can be canceled or measured.

The `newAccessController` is invoked just once for the lifetime of the registry. I don't see a value in passing a context to the access controller upon init. Especially when all the other requests it will deal with will have context of their own - and completely different.